src/core/lombok/eclipse/handlers/HandleEqualsAndHashCode.java

```diff
@@ -238,6 +272,7 @@ public MethodDeclaration createHashCode(EclipseNode type, Collection<Included<Ec
 		method.returnType = TypeReference.baseTypeReference(TypeIds.T_int, 0);
 		setGeneratedBy(method.returnType, source);
 		Annotation overrideAnnotation = makeMarkerAnnotation(TypeConstants.JAVA_LANG_OVERRIDE, source);
+		// TODO: maybe not add this annotation if cacheHashCode is true because we *do* modify a field
```

Comment:
@Rawi01 looking for feedback here as well. We technically modify a field, so not sure we should add this annotation if we are caching the hashCode

Reply:
I checked out the [documentation](https://checkerframework.org/api/org/checkerframework/dataflow/qual/SideEffectFree.html) for this annotation . It seems like caching is allowed but such a method will fail during checks: 
```
The method is allowed to cache the answer to a computationally expensive query, for instance.
```
```
These rules are conservative: any code that passes the checks is side-effect-free, but the Checker 
Framework may issue false positive warnings, for code that uses one of the forbidden constructs 
but is side-effect-free nonetheless. In particular, a method that caches its result will be rejected. 

In fact, the rules are so conservative that checking is currently disabled by default...```

Reply:
That is a weird set of statements on their end. It is allowed, but will fail... it's kinda weird. Checking if it's something the user can disable externally. If not, we probably should remove the annotation.

Reply:
Yes, these statements are confusing :smile:. 

If I understand that stuff correctly they usually [trust these annotations](https://checkerframework.org/manual/#purity-trusted) without actually checking the method body. To prevent mistakes they offer a flag to check if these methods are actually side-effect free by analyzing the method body. Thats what will fail because this check needs some improvements. 

I think that instead of removing the annotation we can actually use `@Pure` for cached hashcodes.

Reply:
Done :wink:

Reply:
Force-push because GitHub thinks `@Pure` in the commit message was me quoting a user with that username.